Microsoft.HealthVault.Fhir/Transformers/DiagnosticReportToHealthVault.cs

```diff
@@ -0,0 +1,253 @@
+using Microsoft.HealthVault.Fhir.FhirExtensions.Helpers;
+using Microsoft.HealthVault.ItemTypes;
+using FhirOrganization = Hl7.Fhir.Model.Organization;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static class DiagnosticReportToHealthVault
+    {
+        public static LabTestResults ToHealthVault(this DiagnosticReport diagnosticReport)
+        {
+            var labTestResults = diagnosticReport.ToThingBase<LabTestResults>();
+
+            if (diagnosticReport.Issued.HasValue)
+            {
+                labTestResults.When = new FhirDateTime(diagnosticReport.Issued.Value).ToAproximateDateTime();
+            }
+
+            var orderByReference = diagnosticReport
+                   .GetExtensionValue<ResourceReference>(HealthVaultExtensions.LabTestResultOrderBy);
+            if (orderByReference != null)
+            {
+                var fhirOrganization = diagnosticReport.GetReferencedResource(orderByReference,
+                     reference => new FhirOrganization { Name = reference.Display });
+                labTestResults.OrderedBy = fhirOrganization.ToHealthVault();
+            }
+
+            foreach (var reference in diagnosticReport.Result)
+            {
+                var labTestResultGroup = GetLabTestResultGroup(diagnosticReport, reference);
+
+                labTestResults.Groups.Add(labTestResultGroup);
+            }
+
+            return labTestResults;
+        }
+
+        private static LabTestResultGroup GetLabTestResultGroup(DiagnosticReport diagnosticReport, ResourceReference reference)
+        {
+            var observation = diagnosticReport.GetContainedResource<Observation>(reference);
+
+            var labTestResultGroup = new LabTestResultGroup { };
+
+            labTestResultGroup.GroupName = observation.Code.ToCodableValue();
+
+            if (observation.Performer.Any())
+            {
+                var fhirOrganization = diagnosticReport.ExtractFirstContainedResource(observation.Performer,
+                     () =>
+                     {
+                         var resource = observation.Performer.FirstOrDefault(performerReference
+                                         => string.IsNullOrEmpty(performerReference.Display));
+                         if (resource != null)
+                         {
+                             return new FhirOrganization
+                             {
+                                 Name = resource.Display
+                             };
+                         }
+                         return null;
+                     });
+                labTestResultGroup.LaboratoryName = fhirOrganization?.ToHealthVault();
+            }
+
+            labTestResultGroup.Status = GetStatus(observation.Status);
+
+            foreach (var relatedObservationLink in observation.Related)
+            {
+                switch (relatedObservationLink.Type)
+                {
+                    case null:
```

Comment:
What's the reasoning behind a null being treated like a `HasMember` value instead of the unsupported values?